tests/river/e2e/api/test_preview.py

```diff
@@ -34,3 +47,57 @@ def test_preview(api_client, export_data, snapshot):
     # - meta.tag.*.code: source_id or resource_id,
     # changes at each run since the mapping is re-imported
     assert fhir_instance == snapshot(exclude=paths("meta.lastUpdated", "meta.tag.0.code", "meta.tag.1.code"))
+
+
+def test_preview_not_found(api_client, patient_mapping, snapshot):
+    url = reverse("preview")
+
+    data = {
+        "resource_id": patient_mapping,
+        "primary_key_values": ["123456789"],  # does not exist
+    }
+    response = api_client.post(url, data, format="json")
+
+    assert response.status_code == 200, response.data
+    assert len(response.data["errors"]) == 0, response.data["errors"]
+    assert len(response.data["instances"]) == 0
+
+
+@patch(
+    "river.services.fhir_api.validate",
+    return_value={
+        "resourceType": "OperationOutcome",
+        "issue": [{"severity": "error", "code": "invalid", "diagnostics": "shit happened"}],
+    },
+)
+def test_preview_with_validation_error(mock_validate, api_client, patient_mapping, snapshot):
+    url = reverse("preview")
+
+    data = {
+        "resource_id": patient_mapping,
+        "primary_key_values": ["10006"],
+    }
+    response = api_client.post(url, data, format="json")
+
+    assert response.status_code == 200, response.data
+    assert len(response.data["errors"]) == 1, response.data["errors"]
+    assert len(response.data["instances"]) == 1
+    operation_outcome = response.data["errors"][0]
+    assert operation_outcome == snapshot()
+
+
+@patch("river.services.fhir_api.validate", side_effect=Exception("fhir-api-returned-an-internal-server-error"))
+def test_preview_with_validation_exception(mock_validate, api_client, patient_mapping, snapshot):
+    url = reverse("preview")
+
+    data = {
+        "resource_id": patient_mapping,
+        "primary_key_values": ["10006"],
+    }
+    response = api_client.post(url, data, format="json")
+
+    assert response.status_code == 200, response.data
+    assert len(response.data["errors"]) == 1, response.data["errors"]
+    assert len(response.data["instances"]) == 1
+    operation_outcome = response.data["errors"][0]
+    assert operation_outcome == snapshot()
```

Comment:
Maybe I'm wrong but I tend to think that the snapshot is a bit overused here
The `operation_outcome` isn't too big and I'm afraid we updated it without really having a look at it